test/unit/logic/transaction.js

```diff
@@ -0,0 +1,1076 @@
+};
+
+var validTransaction = {
+	id: '16140284222734558289',
+	rowId: 133,
+	blockId: '1462190441827192029',
+	type: 0,
+	timestamp: 33363661,
+	senderPublicKey: 'c094ebee7ec0c50ebee32918655e089f6e1a604b83bcaa760293c61e0f18ab6f',
+	senderId: '16313739661670634666L',
+	recipientId: '5649948960790668770L',
+	amount: 8067474861277,
+	fee: 10000000,
+	signature: '7ff5f0ee2c4d4c83d6980a46efe31befca41f7aa8cda5f7b4c2850e4942d923af058561a6a3312005ddee566244346bdbccf004bc8e2c84e653f9825c20be008',
+	signSignature: null,
+	requesterPublicKey: null,
+	signatures: null,
+	asset: {},
+};
+
+var rawValidTransaction = {
+	t_id: '16140284222734558289',
+	b_height: 981,
+	t_blockId: '1462190441827192029',
+	t_type: 0,
+	t_timestamp: 33363661,
+	t_senderPublicKey: 'c094ebee7ec0c50ebee32918655e089f6e1a604b83bcaa760293c61e0f18ab6f',
+	m_recipientPublicKey: null,
+	t_senderId: '16313739661670634666L',
+	t_recipientId: '5649948960790668770L',
+	t_amount: 8067474861277,
+	t_fee: 10000000,
+	t_signature: '7ff5f0ee2c4d4c83d6980a46efe31befca41f7aa8cda5f7b4c2850e4942d923af058561a6a3312005ddee566244346bdbccf004bc8e2c84e653f9825c20be008',
+	confirmations: 8343
+};
+
+var genesisTrs = {
+	type: 0,
+	amount: 10000000000000000,
+	fee: 0,
+	timestamp: 0,
+	recipientId: '16313739661670634666L',
+	senderId: '1085993630748340485L',
+	senderPublicKey: 'c96dec3595ff6041c3bd28b76b8cf75dce8225173d1bd00241624ee89b50f2a8',
+	signature: 'd8103d0ea2004c3dea8076a6a22c6db8bae95bc0db819240c77fc5335f32920e91b9f41f58b01fc86dfda11019c9fd1c6c3dcbab0a4e478e3c9186ff6090dc05',
+	blockId: '9314232245035524467',
+	id: '1465651642158264047'
+};
+
+var validUnconfirmedTrs = {
+	type: 0,
+	amount: 8067474861277,
+	senderPublicKey: 'c094ebee7ec0c50ebee32918655e089f6e1a604b83bcaa760293c61e0f18ab6f',
+	senderId: '16313739661670634666L',
+	requesterPublicKey: null,
+	timestamp: 33641482,
+	asset: {},
+	recipientId: '5649948960790668770L',
+	signature: '24c65ac5562a8ae252aa308926b60342829e82f285e704814d0d3c3954078c946d113aa0bd5388b2c863874e63f71e8e0a284a03274e66c719e69d443d91f309',
+	fee: 10000000,
+	id: '16580139363949197645' 
+};
+
+
+describe('transaction', function () {
+
+	var transaction;
+	var accountModule;
+
+	var attachTransferAsset = function (transaction, accountLogic, rounds, done) {
+		modulesLoader.initModuleWithDb(AccountModule, function (err, __accountModule) {
+			var transfer = new Transfer();
+			transfer.bind(__accountModule, rounds);
+			transaction.attachAssetType(transactionTypes.SEND, transfer);
+			accountModule = __accountModule;
+			done();
+		}, {
+			logic: {
+				account: accountLogic,
+				transaction: transaction
+			}
+		});
+	};
+
+	before(function (done) {
+		async.auto({
+			rounds: function (cb) {
+				modulesLoader.initModule(Rounds, modulesLoader.scope,cb);
+			},
+			accountLogic: function (cb) {
+				modulesLoader.initLogicWithDb(AccountLogic, cb);
+			},
+			transaction: ['accountLogic', function (result, cb) {
+				modulesLoader.initLogicWithDb(Transaction, cb, {
```

Comment:
This object which I pass gets merged default `scope` in initModule here: https://github.com/SargeKhan/lisk/blob/6037bdc5d3daf59eb4f5eeba1d864660f21e49b9/test/common/initModule.js#L208